src/Umbraco.Core/Scoping/Scope.cs

```diff
@@ -226,7 +285,24 @@ public IUmbracoDatabase Database
                 EnsureNotDisposed();
 
                 if (_database != null)
+                {
+                    // If the database has already been resolved, we are already in a
+                    // transaction, but it's possible that more locks have been requested
+                    // so acquire them.
+
+                    // TODO: This is the issue we face with non-eager locks. If locks are
```

Comment:
@nikolajlauridsen IMO it would be good for someone to investigate this TODO change. Can we move the logic to ensure locks are executed to the `UmbracoDatabase` , potentially on the `OnExecutingCommand` method or similar instead of relying on ensuring the locks when this property is accessed? @bergmania might be a good research task for someone. This PR will work and will improve perf quite a lot but this TODO is still relevant.